### PR TITLE
Refactor crates to remove cyclic dependencies

### PR DESCRIPTION
**Pull request by ryangoree**
# Resolved Issues

Related to #92

# Description

This refactors the crates to remove all cyclic dependencies. Below is an illustration of the dependency tree before and after with a few notes:

hyperdrive-rs-dep-trees

# Review Checklists

Please check each item **before approving** the pull request. While going
through the checklist, it is recommended to leave comments on items that are
referenced in the checklist to make sure that they are reviewed.

- [ ] **Testing**
    - [ ] Are there new or updated unit or integration tests?
    - [ ] Do the tests cover the happy paths?
    - [ ] Do the tests cover the unhappy paths?
    - [ ] Are there an adequate number of fuzz tests to ensure that we are
          covering the full input space?
    - [ ] If matching Solidity behavior, are there differential fuzz tests that
          ensure that Rust matches Solidity?

### CODE REVIEW

File: crates/hyperdrive-wrappers/build.rs [hunk 1]
```diff
@@ -99,13 +101,14 @@ fn main() -> Result<()> {
     }
 
     // Compile the contracts.
+    // Command::new("forge")
```
Comment (sentilesdal) [inline crates/hyperdrive-wrappers/build.rs]: dead code?

File: crates/fixed-point/src/lib.rs [hunk 2]
```diff
@@ -555,7 +553,7 @@ mod tests {
 
         // Fuzz the rust and solidity implementations against each other.
         let mut rng = thread_rng();
-        for _ in 0..*FAST_FUZZ_RUNS {
+        for _ in 0..1000 {
```
Comment (sentilesdal) [inline crates/fixed-point/src/lib.rs:570]: leaving as 1000 on purpose?

File: crates/hyperdrive-test-utils/src/chain/deploy.rs [hunk 3]
```diff
@@ -1042,6 +1060,8 @@ mod tests {
         chain.deal(ALICE.address(), uint256!(100_000e18)).await?;
         let client = chain.client(ALICE.clone()).await?;
 
+        println!("HERE");
```
Comment (sentilesdal) [inline crates/hyperdrive-test-utils/src/chain/deploy.rs:1063]: println

File: crates/hyperdrive-test-utils/src/constants.rs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+use std::env;
+
+pub use test_utils::constants::*;
+
+lazy_static! {
```
Comment (sentilesdal) [inline crates/hyperdrive-test-utils/src/constants.rs:5]: very nice, i need to do this in some of the lp files.

File: crates/hyperdrive-test-utils/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#![allow(clippy::comparison_chain)]
```
Comment (sentilesdal) [inline crates/hyperdrive-test-utils/src/lib.rs:1]: ooo what's this?

**Review by sentilesdal — APPROVED** [review]
this is amazing, nice work.

File: crates/fixed-point/src/lib.rs (hunk 2 above)
Comment (ryangoree) [reply crates/fixed-point/src/lib.rs:570]: Nope! Good catch

File: crates/hyperdrive-test-utils/src/lib.rs (hunk 5 above)
Comment (ryangoree) [reply crates/hyperdrive-test-utils/src/lib.rs:1]: This is left over from before. It prevents clippy from complaining about using `if` statements in comparison chains that could be rewritten with `match` or `cmp`, but I just rant lint without it locally and it passed, so I don't think it's needed anymore 👍

Comment (sentilesdal) [reply crates/hyperdrive-test-utils/src/lib.rs]: oh yeah, Ive seen those warnings, they're annoying.

**Pull request merged**